index.js

```diff
@@ -1093,6 +1093,10 @@ Primus.prototype.reserved.events = {
  */
 Primus.createSocket = function createSocket(options) {
   options = options || {};
+  // Make sure the temporary Primus we create below doesn't start a heartbeat
+  options.pingInterval = 'pingInterval' in options
```

Comment:
Should we set it to `false` unconditionally? Can't think of a case where starting the heartbeat timer makes sense here.